pkg/gofr/http/responder_test.go

```diff
@@ -207,6 +208,122 @@ func TestResponder_TemplateResponse(t *testing.T) {
 	assert.Equal(t, expectedBody, responseBody)
 }
 
+func TestResponder_CustomErrorWithResponse(t *testing.T) {
+	w := httptest.NewRecorder()
+	responder := NewResponder(w, http.MethodGet)
+
+	customErr := &CustomError{
+		Code:    http.StatusNotFound,
+		Message: "resource not found",
+		Title:   "Custom Error",
+	}
+
+	responder.Respond(nil, customErr)
+
+	resp := w.Result()
+	defer resp.Body.Close()
+
+	assert.Equal(t, http.StatusNotFound, resp.StatusCode)
+
+	var body map[string]any
+	err := json.NewDecoder(resp.Body).Decode(&body)
+	require.NoError(t, err)
+
+	errorObj := body["error"].(map[string]any)
+
+	assert.Equal(t, "resource not found", errorObj["message"])
+	assert.Equal(t, int(http.StatusNotFound), int(errorObj["code"].(float64)))
```

Comment:
You can use EqualValues here


```suggestion
	assert.EqualValues(t, http.StatusNotFound, errorObj["code"])
```

Reply:
sure done